test/fail_compilation/bug9631.d

```diff
@@ -58,3 +58,54 @@ void test3()
     S[1] sa = cast(S[1])ta;
     auto t2 = cast(tem!().S[])sa;
 }
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/bug9631.d(79): Error: function bug9631.arg.f `(int i, S s)` is not callable using argument types `(int, S)`
+fail_compilation/bug9631.d(79):        cannot pass argument `y` of type `bug9631.tem!().S` to parameter `S s` of type `bug9631.S`
```

Comment:
> function bug9631.arg.f \`(int i, S s)\` is not callable using argument types \`(int, S)\`

1. Tick marks should surround the entire function declaration (e.g \`bug9631.arg.f (int i, S s)\`).  That may be a difficult thing to do and I'd be willing to pass on the tick marks for now
2.  The function accepts `(int, S)` but the error message says it's not callable using `(int, S)`.  Does this need fully qualified names or something?

Reply:
> cannot pass argument `y` of type `bug9631.tem!().S` to parameter `S s` of type `bug9631.S`

I suspect this line is supposed the clarify the first, but it's awfully confusing.  Can't we have just one line with fully qualified type names?

Reply:
> bug9631.arg.f `(int i, S s)`

This was introduced by Walter before this PR. I tried to fix it but can't work out why exactly it happens.